pyanaconda/modules/users/users.py

```diff
@@ -215,10 +218,17 @@ def _ksdata_to_user_data(self, user_ksdata):
         # -> as user data is -1 by default ve only set it if kickstart has something,
         #    that is not None
         # We need to make sure we correctly convert between these two.
+        # We also cannot use values larger then the accepted range.
         if user_ksdata.uid is not None:
-            user_data.uid = user_ksdata.uid
+            user_data.uid = min(user_ksdata.uid, MAX_ID_VALUE)
+            user_data.uid_status = ""
+        else:
+            user_data.uid_status = ID_NOT_SET
         if user_ksdata.gid is not None:
-            user_data.gid = user_ksdata.gid
+            user_data.gid = min(user_ksdata.gid, MAX_ID_VALUE)
+            user_data.gid_status = ""
```

Comment:
So the status will be an empty string or `not set`. I don't know what I would expect from a status that doesn't tell me anything. And I have no idea how to interpret a configuration with `gid` set to `12345` and `gid_status` set to `not_set`.

First of all, I would like to avoid `status` in the property name. I think it has a little different meaning. What do you think about `uid_mode` and `gid_mode`?

Second, the property has to use meaningful values. I would suggest `USE_DEFAULT` and `USE_VALUE` (or just `DEFAULT` and `VALUE`). When `gid` is not set, then `groupadd` chooses a reasonable default. It works the same way with `uid` and `useradd`.

Reply:
Thank you, I did just that.